src/routers/xcm/Router.sol

```diff
@@ -61,27 +125,75 @@ contract ConnectorXCMRouter {
             (uint64 poolId, bytes16 trancheId, uint128 price) = ConnectorMessages.parseUpdateTokenPrice(_msg);
             connector.updateTokenPrice(poolId, trancheId, price);
         } else if (ConnectorMessages.isTransfer(_msg)) {
-            (uint64 poolId, bytes16 trancheId,, address user, uint256 amount) = ConnectorMessages.parseTransfer(_msg);
+            (uint64 poolId, bytes16 trancheId,, address user, uint128 amount) = ConnectorMessages.parseTransfer(_msg);
             connector.handleTransfer(poolId, trancheId, user, amount);
         } else {
             require(false, "invalid-message");
         }
     }
 
-    function sendMessage(uint64 poolId, bytes16 trancheId, uint256 amount, address user) external onlyConnector {
-        // TODO: implement
+    function send(bytes memory message) public onlyConnector {
+        bytes memory centChainCall = centrifuge_handle_call(message);
+
+        XCM_TRANSACTOR_V2_CONTRACT.transactThroughSignedMultilocation(
+            // dest chain
+            centrifuge_parachain_multilocation(),
+            // fee asset
+            cfg_asset_multilocation(),
+            // the weight limit for the transact call execution
+            xcmWeightInfo.transactWeightAtMost,
+            // the call to be executed on the cent chain
+            centChainCall,
+            // the CFG we offer to pay for execution fees of the whole XCM
+            xcmWeightInfo.feeAmount,
+            // overall XCM weight, the total weight the XCM-transactor extrinsic can use.
+            // This includes all the XCM instructions plus the weight of the Transact call itself.
+            xcmWeightInfo.buyExecutionWeightLimit
+        );
     }
 
-    function bytes32ToString(bytes32 _bytes32) internal pure returns (string memory) {
-        uint8 i = 0;
-        while (i < 32 && _bytes32[i] != 0) {
-            i++;
-        }
+    function centrifuge_handle_call(bytes memory message) internal view returns (bytes memory) {
+        return abi.encodePacked(
+            // The call index; first byte is the pallet, the second is the extrinsic
+            centrifugeChainHandleCallIndex,
+            // We need to specify the length of the message in the scale-encoding format
+            message_length_scale_encoded(message),
+            // The connector message itself
+            message
+        );
+    }
```

Comment:
What does this view expose?

Reply:
`view` means that this function reads storage values, in this case `centrifugeChainHandleCallIndex`